### PR TITLE
Liquid::TablerowloopDrop

### PR DESCRIPTION
**Pull request by fw42**
Same as https://github.com/Shopify/liquid/pull/681 but for table rows.

@pushrax @dylanahsmith

### CODE REVIEW

File: lib/liquid/tags/table_row.rb [hunk 1]
```diff
@@ -28,36 +28,21 @@ def render(context)
...
-            'col'.freeze       => col + 1,
-            'col0'.freeze      => col,
-            'rindex'.freeze    => length - index,
-            'rindex0'.freeze   => length - index - 1,
-            'first'.freeze     => (index == 0),
-            'last'.freeze      => (index == length - 1),
-            'col_first'.freeze => (col == 0),
-            'col_last'.freeze  => (col == cols - 1)
-          }
+          context['tablerowloop'.freeze] = tablerowloop
```
Comment (dylanahsmith) [inline lib/liquid/tags/table_row.rb]: Shouldn't this be moved out of the loop now that we are re-using the object?

Comment (fw42) [inline lib/liquid/tags/table_row.rb]: good point, yeah

**Comment by dylanahsmith** [comment]
:+1:

File: lib/liquid/tags/table_row.rb [hunk 2]
```diff
@@ -28,36 +28,21 @@ def render(context)
...
-
-          if col == cols && (index != length - 1)
-            col  = 0
-            row += 1
-            result << "</tr>\n<tr class=\"row#{row}\">"
+          if tablerowloop.col_last && !tablerowloop.last
+            result << "</tr>\n<tr class=\"row#{tablerowloop.row + 1}\">"
           end
+
+          tablerowloop.send(:increment!)
```
Comment (parkr) [inline lib/liquid/tags/table_row.rb]: why use `send` here?

Comment (fw42) [inline lib/liquid/tags/table_row.rb]: because the method isn't `public` (if it were, people could call it from within the template, which we don't want).

**Comment by pushrax** [comment]
:ship:  I wish we didn't have duplication between for and tablerow, but oh well.

**Pull request merged**